src/js/components/NewConnectionModal.tsx

```diff
@@ -110,22 +111,25 @@ export default function NewConnectionModal() {
           return obj
         }, {})
         try {
+          setIsFetching(true)
           await dispatch(setConnection(toCluster({host})))
         } catch (_) {
           setErrors([{message: "Cannot connect to host"}])
           return
+        } finally {
+          setIsFetching(false)
         }
         closeModal()
       } else {
         setErrors(form.getErrors())
       }
     },
-    [f, config]
+    [f, config, setIsFetching]
   )
 
   const buttons = [
     {label: "Cancel", click: (closeModal) => closeModal()},
-    {label: "Connect", click: onSubmit}
+    {label: "Connect", click: onSubmit, showSpinner: isFetching}
```

Comment:
Instead of adding a `showSpinner` option, why not add a more generic "icon" option where you could inject the MacSpinner.

This does make me think about buttons with a loading state. I'd like to design a first-class "loading" state into the buttons. I always feel comforted as a user when I see those in other apps.